### DISCUSSION TITLE
Setting up git-gateway without netlify

### DISCUSSION THREAD

**Discussion by SChetwynd** [StaticJsCMS/static-cms#995] [Q&A]
Hi,

I'm trying to setup static-cms to use git-gateway without netlify, I've got a GoTrue Git-Gateway up an running, but I cannot find any documentation on how to link these to static-cms. The docs say:

> To configure in Static CMS, use the same backend settings in your Static CMS config file as described in Step 2 of the [Git Gateway with Netlify Identity](https://v1.staticcms.org/docs/git-gateway-backend#git-gateway-with-netlify-identity) instructions above.
>
> &mdash; <cite>[Static CMS docs](https://v1.staticcms.org/docs/git-gateway-backend#git-gateway-without-netlify)<cite>

But step to is simply to include:

```yml
backend:
  name: git-gateway
```

When attempting to login with this set up I get this error:

> Unable to access identity settings. When using git-gateway backend make sure to enable Identity service and Git Gateway.

And I see in the network tab the browser is attempting to access:  `https://www.<My Blog>.co.uk/.netlify/identity/settings` (I've replaced the name of my website with `<My Blog>` the url does not literally include `<My Blog>` :smile:  ) which doesn't exist.

I've also attempted this with the `backend` setup with working gitlab details, but with the `name` set to `git-gateway` but that gets the same result.

I'm running a Hugo site, which has never been on netlify, and using gitlab as the git provider, not sure if that'll make and difference.

Any help would be greatly appreciated.

**Comment by mnordhaus** [comment]
Hi @SChetwynd 

I try to get it up and running as well but have not find the time yet to finish it. However:

Have you tried the following settings:

```yaml
backend:
  name: git-gateway
  branch: main
  identity_url: "your url to your Go True"
  gateway_url:  "the URL to your Git Gateway"
  site_domain:  "your domain of your website"
```

It should as far as I understand also work with the GitHub backend: https://www.staticcms.org/docs/github-backend.

Please let me know how if it works.

Best,

Markus

**Reply by SChetwynd** [reply]
Hi @mnordhaus ,

Thanks for the response, I've given that a go, but its still not working I'm getting the same results as I mentioned above.

I forgot to mention above that my admin.html looks like this:

```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <link rel="stylesheet" href="https://unpkg.com/@staticcms/app@^3.0.0/dist/main.css" />
    <title>Content Manager</title>
  </head>
  <body>
    <!-- Include the script that builds the page and powers Static CMS -->
    <script src="https://identity.netlify.com/v1/netlify-identity-widget.js"></script>
    <script src="https://unpkg.com/@staticcms/app@^3.0.0/dist/static-cms-app.js"></script>
    <script>
      window.CMS.init();
    </script>
  </body>
</html>
```

I'm using the netlify identity widget, is that correct? I couldn't find any docs about using a different widget.

Thanks
Steven

**Reply by mnordhaus** [reply]
I guess you need to host the widget locally: https://github.com/netlify/netlify-identity-widget

**Reply by mnordhaus** [reply]
How does your configuration look regarding the identity and gateway url?

**Reply by SChetwynd** [reply]
I've discovered that I can use the netlify identity widget, their docs say it can be initialized with a custom APIUrl and doing that I was able to use my instance of gotrue, like so:

```javascript
    <script src="https://identity.netlify.com/v1/netlify-identity-widget.js"></script>
    <script>
      netlifyIdentity.init({
        APIUrl: "my gotrue url"
      });
    </script>
    <script src="https://unpkg.com/@staticcms/app@^3.0.0/dist/static-cms-app.js"></script>
```

But thats as far as I've got, the gotrue docker container I'm using is causing me some issues, which I've not had time to sort out.

**Comment by mnordhaus** [comment]
Hi @SChetwynd 

Have you already seen this one? That is a documentation that has been created a while ago, I would like to bring it up to date.

https://github.com/mnordhaus/netlify-cms-with-selfhosted-gotrue-and-git-gateway

I hope that helps, and I hope that this discussion leads to a good documentation. I see a need for having the netlify products be forked and managed by other maintainers. Like @KaneFreeman did it for netlifycms. Maybe we can spark the fire here too?

Best,

Markus

**Comment by mnordhaus** [comment]
@SChetwynd Hey, did you have any success on this topic?
